esphome/components/libretiny_pwm/output.py

```diff
@@ -0,0 +1,74 @@
+from esphome import pins, automation
+from esphome.components import output
+import esphome.config_validation as cv
+import esphome.codegen as cg
+from esphome.const import (
+    CONF_FREQUENCY,
+    CONF_ID,
+    CONF_PIN,
+)
+
+CONFLICTS_WITH = ["esp8266", "esp32", "rp2040"]
+
+
+def calc_max_frequency(bit_depth):
+    return 80e6 / (2**bit_depth)
+
+
+def calc_min_frequency(bit_depth):
+    max_div_num = ((2**20) - 1) / 256.0
+    return 80e6 / (max_div_num * (2**bit_depth))
+
+
+def validate_frequency(value):
```

Comment:
I'm not sure if we need frequency validation yet, we don't even know what the allowed frequencies are.